homeassistant/components/withings/__init__.py

```diff
@@ -0,0 +1,112 @@
+"""
+Support for the Withings API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/withings/
+"""
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+from homeassistant.helpers import config_validation as cv
+
+from . import config_flow, const
+from .common import _LOGGER, create_withings_data_manger, NotAuthenticatedError
+from .sensor import WITHINGS_MEASUREMENTS_MAP
+
+DOMAIN = const.DOMAIN
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(const.CLIENT_ID): vol.All(cv.string, vol.Length(min=1)),
+                vol.Required(const.CLIENT_SECRET): vol.All(
+                    cv.string, vol.Length(min=1)
+                ),
+                vol.Optional(const.BASE_URL): cv.url,
+                vol.Required(const.PROFILES): vol.All(
+                    cv.ensure_list,
+                    vol.Unique(),
+                    vol.Length(min=1),
+                    [vol.All(cv.string, vol.Length(min=1))],
+                ),
+                vol.Optional(
+                    const.MEASURES, default=list(WITHINGS_MEASUREMENTS_MAP)
+                ): vol.All(
+                    cv.ensure_list,
+                    vol.Unique(),
+                    vol.Length(min=1),
+                    [vol.All(cv.string, vol.In(list(WITHINGS_MEASUREMENTS_MAP)))],
+                ),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType):
+    """Set up the Withings component."""
+    conf = config.get(DOMAIN)
+    if not conf:
+        return True
+
+    hass.data[DOMAIN] = {const.CONFIG: conf}
+
+    base_url = conf.get(const.BASE_URL, hass.config.api.base_url).rstrip("/")
+
+    hass.http.register_view(config_flow.WithingsAuthCallbackView)
+
+    config_flow.register_flow_implementation(
+        hass,
+        conf.get(const.CLIENT_ID),
+        conf.get(const.CLIENT_SECRET),
+        base_url,
+        conf.get(const.PROFILES),
+    )
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": const.SOURCE_USER}, data={}
```

Comment:
This should be `SOURCE_IMPORT`.

Reply:
Does that mean 'async_step_user' should also be renamed to 'async_step_import'?

Reply:
No, the user and import steps are different entry points to the config flow. It is common to pass the data from the import step directly to the user step, if no extra validation is required in the import step.